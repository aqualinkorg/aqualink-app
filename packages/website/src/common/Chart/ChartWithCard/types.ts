export type RangeValue = "three_months" | "one_year" | "max" | "custom";

export interface RangeButton {
  id: RangeValue;
  title: string;
  disabled?: boolean;
  tooltip: string;
}

export interface CardColumn {
  title: string;
  key: string;
  color: string;
  rows: { key: string; value: number | undefined }[];
  tooltip?: string;
  display: boolean;
}

export type Dataset = "spotter" | "hobo" | "sst";
