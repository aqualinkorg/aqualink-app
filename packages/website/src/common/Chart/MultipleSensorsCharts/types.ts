import { DataRange, ValueWithTimestamp } from "../../../store/Sites/types";

export type RangeValue = "one_month" | "one_year" | "max" | "custom";

export interface RangeButton {
  id: RangeValue;
  title: string;
  disabled?: boolean;
  tooltip: string;
}

export interface CardColumn {
  title: string;
  key: string;
  color: string;
  rows: { key: string; value: number | undefined }[];
  tooltip?: string;
  display: boolean;
  unit: string;
}

export interface OceanSenseDataset {
  unit: string;
  data: ValueWithTimestamp[];
  title: string;
  id: string;
}

export interface AvailableRange {
  name: string;
  data?: DataRange[];
}
