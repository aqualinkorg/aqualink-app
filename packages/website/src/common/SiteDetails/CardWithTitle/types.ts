import { TypographyProps } from "@material-ui/core";
import { CSSProperties } from "@material-ui/core/styles/withStyles";

export interface Value {
  text: string;
  variant: TypographyProps["variant"];
  marginRight: CSSProperties["marginRight"];
  overflowEllipsis?: boolean;
}
