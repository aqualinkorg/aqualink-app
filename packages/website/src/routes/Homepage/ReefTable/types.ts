export type Order = "asc" | "desc" | undefined;

export type OrderKeys = "locationName" | "temp" | "depth" | "dhw";
