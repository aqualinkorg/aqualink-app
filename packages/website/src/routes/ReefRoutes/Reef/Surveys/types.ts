export interface EditPoiNameDraft {
  [key: number]: string | null;
}
