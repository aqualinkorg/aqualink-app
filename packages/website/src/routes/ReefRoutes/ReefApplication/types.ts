export interface AgreementsChecked {
  shipping: boolean;
  buoy: boolean;
  survey: boolean;
}
