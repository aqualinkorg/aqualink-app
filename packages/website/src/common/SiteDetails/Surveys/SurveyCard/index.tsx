import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Paper,
  Grid,
  CardMedia,
  Typography,
  Button,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import { formatNumber } from "../../../../helpers/numberUtils";
import DeleteButton from "../DeleteButton";
import { SurveyListItem } from "../../../../store/Survey/types";
import incomingStyles from "../styles";
import CustomLink from "../../../Link";

const SurveyCard = ({
  pointId,
  pointName,
  isAdmin,
  siteId,
  survey,
  classes,
}: SurveyCardProps) => {
  const isShowingFeatured = pointId === -1;

  return (
    <Paper elevation={0} className={classes.surveyCard}>
      <Grid style={{ height: "100%" }} container justify="space-between">
        <Grid className={classes.cardImageWrapper} item xs={12} md={5}>
          <Link to={`/sites/${siteId}/survey_details/${survey.id}`}>
            <CardMedia
              className={classes.cardImage}
              image={
                isShowingFeatured
                  ? survey.featuredSurveyMedia?.url
                  : survey.surveyPointImage?.[pointId]?.[0]
              }
            />
          </Link>
        </Grid>
        <Grid className={classes.infoWrapper} container item xs={12} md={7}>
          <Grid className={classes.info} container item xs={12}>
            {survey.user?.fullName && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  User:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {survey.user.fullName}
                </Typography>
              </Grid>
            )}
            {survey.featuredSurveyMedia?.surveyPoint?.name && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Survey Point:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin} ${classes.blueText}`}
                  variant="h6"
                  title={survey.featuredSurveyMedia.surveyPoint.name}
                >
                  <CustomLink
                    to={`/sites/${siteId}/points/${
                      isShowingFeatured
                        ? survey.featuredSurveyMedia.surveyPoint.id
                        : pointId
                    }`}
                    isIcon={false}
                    tooltipTitle=""
                    content={
                      pointName === "All"
                        ? survey.featuredSurveyMedia.surveyPoint.name
                        : pointName
                    }
                  />
                </Typography>
              </Grid>
            )}
            {survey.comments && (
              <Grid
                className={classes.commentsWrapper}
                container
                alignItems="flex-start"
                item
                xs={12}
              >
                <Grid style={{ height: "80%" }} item xs={12}>
                  <Typography className={classes.cardFields} variant="h6">
                    Comments:
                  </Typography>
                  <Typography
                    className={`${classes.cardValues} ${classes.comments}`}
                    variant="h6"
                  >
                    {survey.comments}
                  </Typography>
                </Grid>
              </Grid>
            )}
            {survey.temperature && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Temp:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {`${formatNumber(survey.temperature, 1)} Â°C`}
                </Typography>
              </Grid>
            )}
            <Grid
              container
              alignItems="center"
              justify="space-between"
              item
              xs={12}
            >
              <Grid item xs={10}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/sites/${siteId}/survey_details/${survey.id}`}
                >
                  <Button size="small" variant="outlined" color="primary">
                    VIEW DETAILS
                  </Button>
                </Link>
              </Grid>
              {isAdmin && (
                <Grid container justify="flex-end" item xs={2}>
                  <DeleteButton
                    siteId={siteId}
                    surveyId={survey.id}
                    diveDate={survey.diveDate}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    ...incomingStyles,
    cardImageWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    infoWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    commentsWrapper: {
      maxHeight: "51%",
    },
    comments: {
      height: "100%",
      overflowY: "auto",
    },
    valuesWithMargin: {
      marginLeft: "1rem",
      maxWidth: "60%",
      display: "block",
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
    },
    blueText: {
      color: theme.palette.primary.main,
    },
    info: {
      height: "100%",
      padding: "0.5rem 0.5rem 0.5rem 1rem",
    },
  });

interface SurveyCardIncomingProps {
  pointId: number;
  pointName: string | null;
  isAdmin: boolean;
  siteId: number;
  survey: SurveyListItem;
}

type SurveyCardProps = SurveyCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyCard);
