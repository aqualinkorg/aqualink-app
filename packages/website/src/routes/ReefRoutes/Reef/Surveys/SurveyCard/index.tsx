import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Paper,
  Grid,
  CardMedia,
  Typography,
  Button,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import { formatNumber } from "../../../../../helpers/numberUtils";
import DeleteButton from "../DeleteButton";
import { SurveyListItem } from "../../../../../store/Survey/types";
import incomingStyles from "../styles";

const SurveyCard = ({ isAdmin, reefId, survey, classes }: SurveyCardProps) => {
  return (
    <Paper elevation={0} className={classes.surveyCard}>
      <Grid style={{ height: "100%" }} container justify="space-between">
        <Grid className={classes.cardImageWrapper} item xs={12} md={5}>
          {survey.featuredSurveyMedia && (
            <CardMedia
              className={classes.cardImage}
              image={survey.featuredSurveyMedia.url}
            />
          )}
        </Grid>
        <Grid className={classes.infoWrapper} container item xs={12} md={7}>
          <Grid className={classes.info} container item xs={12}>
            {survey.userId!.fullName && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  User:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {survey.userId!.fullName}
                </Typography>
              </Grid>
            )}
            {survey.comments && (
              <Grid
                className={classes.commentsWrapper}
                container
                alignItems="flex-start"
                item
                xs={12}
              >
                <Grid style={{ height: "80%" }} item xs={12}>
                  <Typography className={classes.cardFields} variant="h6">
                    Comments:
                  </Typography>
                  <Typography
                    className={`${classes.cardValues} ${classes.comments}`}
                    variant="h6"
                  >
                    {survey.comments}
                  </Typography>
                </Grid>
              </Grid>
            )}
            {survey.temperature && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Temp:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {`${formatNumber(survey.temperature, 1)} Â°C`}
                </Typography>
              </Grid>
            )}
            <Grid
              container
              alignItems="center"
              justify="space-between"
              item
              xs={12}
            >
              <Grid item xs={10}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/reefs/${reefId}/survey_details/${survey.id}`}
                >
                  <Button size="small" variant="outlined" color="primary">
                    VIEW DETAILS
                  </Button>
                </Link>
              </Grid>
              {isAdmin && (
                <Grid container justify="flex-end" item xs={2}>
                  <DeleteButton
                    reefId={reefId}
                    surveyId={survey.id}
                    diveDate={survey.diveDate}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    ...incomingStyles,
    cardImageWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    infoWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    commentsWrapper: {
      maxHeight: "55%",
    },
    comments: {
      height: "100%",
      overflowY: "auto",
    },
    info: {
      height: "100%",
      padding: "0.5rem 0.5rem 0.5rem 1rem",
    },
  });

interface SurveyCardIncomingProps {
  isAdmin: boolean;
  reefId: number;
  survey: SurveyListItem;
}

type SurveyCardProps = SurveyCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyCard);
