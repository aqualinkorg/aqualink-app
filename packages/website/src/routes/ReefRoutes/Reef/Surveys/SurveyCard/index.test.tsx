import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import SurveyCard from ".";
import { SurveyListItem } from "../../../../../store/Survey/types";

const mockStore = configureStore([]);

describe("Surveys Delete Button", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const survey: SurveyListItem = {
      id: 0,
      diveLocation: null,
      diveDate: null,
      weatherConditions: "calm",
      comments: "",
      temperature: 24.5,
      observations: ["possible-disease"],
      userId: {
        id: 0,
        fullName: "Test User",
      },
    };

    const store = mockStore({
      user: {
        userInfo: {
          email: "test@mail.com",
          fullName: "Test User",
          adminLevel: "super_admin",
          firebaseUid: "RaNdOmStRiNg",
          administeredReefs: [],
          token: "RaNdOmStRiNg",
        },
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <SurveyCard isAdmin reefId={0} survey={survey} />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
