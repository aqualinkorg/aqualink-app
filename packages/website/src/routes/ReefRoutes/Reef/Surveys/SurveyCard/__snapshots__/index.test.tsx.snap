// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Surveys Delete Button should render with given state from Redux store 1`] = `
<div>
  <div
    class="MuiPaper-root SurveyCard-surveyCard-5 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-space-between"
      style="height: 100%;"
    >
      <div
        class="MuiGrid-root SurveyCard-cardImageWrapper-10 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      />
      <div
        class="MuiGrid-root SurveyCard-infoWrapper-11 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div
          class="MuiGrid-root SurveyCard-info-14 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
          >
            <mock-typography
              classname="SurveyCard-cardFields-7"
              variant="h6"
            >
              User:
            </mock-typography>
            <mock-typography
              classname="SurveyCard-cardValues-8 SurveyCard-valuesWithMargin-9"
              variant="h6"
            >
              Test User
            </mock-typography>
          </div>
          
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12"
          >
            <mock-typography
              classname="SurveyCard-cardFields-7"
              variant="h6"
            >
              Temp:
            </mock-typography>
            <mock-typography
              classname="SurveyCard-cardValues-8 SurveyCard-valuesWithMargin-9"
              variant="h6"
            >
              24.5 Â°C
            </mock-typography>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
            >
              <a
                href="/reefs/0/survey_details/0"
                style="text-decoration: none;"
              >
                <mock-button
                  color="primary"
                  size="small"
                  variant="outlined"
                >
                  VIEW DETAILS
                </mock-button>
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-2"
            >
              <mock-iconbutton>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </mock-iconbutton>
              <mock-dialog
                open="false"
              >
                <mock-dialogtitle
                  classname="Dialog-dialogTitle-16"
                >
                  <mock-typography>
                    Are you sure you would like to delete the survey for Invalid date? It will delete all media assosciated with this survey.
                  </mock-typography>
                </mock-dialogtitle>
                <mock-dialogcontent>
                  <div
                    class="MuiCollapse-container DeleteButton-alert-15 MuiCollapse-hidden"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
                          role="alert"
                        >
                          <div
                            class="MuiAlert-icon"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                          </div>
                          <div
                            class="MuiAlert-message"
                          />
                          <div
                            class="MuiAlert-action"
                          >
                            <mock-iconbutton
                              color="inherit"
                              size="small"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                />
                              </svg>
                            </mock-iconbutton>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </mock-dialogcontent>
                <mock-dialogactions>
                  <mock-button
                    color="secondary"
                    size="small"
                    variant="contained"
                  >
                    No
                  </mock-button>
                  <mock-button
                    color="primary"
                    size="small"
                    variant="contained"
                  >
                    Yes
                  </mock-button>
                </mock-dialogactions>
              </mock-dialog>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
